From: Carlos Garnacho <carlosg@gnome.org>
Date: Thu, 5 Sep 2019 12:59:08 +0200
Subject: x11: Use the currently focused X window for _NET_ACTIVE_WINDOW

MetaDisplay and MetaX11Display focus windows are slightly decoupled,
we cannot rely here on the MetaDisplay focus to be updated yet. We
however know the X Window that got focused, so lookup the corresponding
MetaWindow (and client X window) from it.

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/mutter/+bug/1842971
Closes: https://gitlab.gnome.org/GNOME/mutter/issues/751
Origin: https://gitlab.gnome.org/GNOME/mutter/merge_requests/776/
Applied-Upstream: 3.34.1
---
 src/x11/meta-x11-display.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/x11/meta-x11-display.c b/src/x11/meta-x11-display.c
index 4645c57..5f2e71c 100644
--- a/src/x11/meta-x11-display.c
+++ b/src/x11/meta-x11-display.c
@@ -1828,12 +1828,15 @@ meta_x11_display_increment_event_serial (MetaX11Display *x11_display)
 void
 meta_x11_display_update_active_window_hint (MetaX11Display *x11_display)
 {
-  MetaWindow *focus_window = x11_display->display->focus_window;
+  MetaWindow *focus_window;
   gulong data[1];
 
   if (x11_display->display->closing)
     return; /* Leave old value for a replacement */
 
+  focus_window = meta_x11_display_lookup_x_window (x11_display,
+                                                   x11_display->focus_xwindow);
+
   if (focus_window)
     data[0] = focus_window->xwindow;
   else
